src/community/wfs-templating/src/main/java/org/geoserver/wfstemplating/request/JsonTemplateCallBackOGC.java

```diff
@@ -0,0 +1,160 @@
+                        || format.equals(TemplateIdentifier.GEOJSON.getOutputFormat()))) {
+            return format;
+        } else if (accept != null) {
+            if (accept.contains(TemplateIdentifier.GEOJSON.getOutputFormat()))
+                return TemplateIdentifier.GEOJSON.getOutputFormat();
+            else if (accept.contains(TemplateIdentifier.JSONLD.getOutputFormat()))
+                return TemplateIdentifier.JSONLD.getOutputFormat();
+        }
+        return null;
+    }
+
+    private void replaceJsonLdPathWithFilter(
+            String strFilter, String outputFormat, Operation operation) throws Exception {
+        if (strFilter != null && strFilter.indexOf(".") != -1) {
+            FeatureTypeInfo typeInfo = getFeatureType((String) operation.getParameters()[0]);
+            RootBuilder root = configuration.getTemplate(typeInfo, outputFormat);
+            if (root != null) {
+                JsonPathVisitor visitor = new JsonPathVisitor(typeInfo.getFeatureType());
+                /* Todo find a better way to replace json-ld path with corresponding template attribute*/
+                // Get filter from string in order to make it accept the visitor
+                Filter f = (Filter) XCQL.toFilter(strFilter).accept(visitor, root);
+                // Taking back a string from Function cause
+                // OGC API get a string cql filter from query string
+                String newFilter =
+                        JsonLdCQLManager.removeQuotes(ECQL.toCQL(f)).replaceAll("/", ".");
+                newFilter = JsonLdCQLManager.quoteXpathAttribute(newFilter);
+                for (int i = 0; i < operation.getParameters().length; i++) {
+                    Object p = operation.getParameters()[i];
+                    if (p != null && ((String.valueOf(p)).trim().equals(strFilter.trim()))) {
+                        operation.getParameters()[i] = newFilter;
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public Response responseDispatched(
+            Request request, Operation operation, Object result, Response response) {
+        String format = getFormatSupportingTemplating(request);
+        if (format != null && format.equals(TemplateIdentifier.GEOJSON.getOutputFormat())) {
+            FeatureTypeInfo typeInfo = null;
+            typeInfo = getFeatureType((String) operation.getParameters()[0]);
+            if (typeInfo != null) {
+                try {
+                    RootBuilder root = configuration.getTemplate(typeInfo, format);
+                    if (root != null) {
+                        GeoJSONGetFeatureResponse featureResponse =
+                                (GeoJSONGetFeatureResponse)
+                                        GeoServerExtensions.bean("geoJSONGetFeatureResponse");
+                        GeoJsonTemplateGetFeatureResponse templatingResp =
```

Comment:
Unsure what's happening here... seems like a workaround for an issue with response resolution? Ah, I'm guessing the dispatcher would not be happy to see two responses producing the same output for the same request and mime type, and it would conflict with the main GeoJSON response?

Reply:
The code here is not setting the outputformat to the serveletResponse. https://github.com/geoserver/geoserver/blob/f81df44a09cde0ccfb249866796d2aaa8d7beba7/src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIBodyMethodProcessor.java#L137